src/Raven.Server/Utils/DirectoryExecUtils.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Raven.Server.Config.Categories;
+using Sparrow.Logging;
+using Sparrow.Utils;
+using Voron;
+
+namespace Raven.Server.Utils
+{
+    public static class DirectoryExecUtils
+    {
+        public static void SubscribeToOnDirectoryExec(StorageEnvironmentOptions options, StorageConfiguration config, string databaseName, EnvironmentType envType, Logger logger)
+        {
+            if (string.IsNullOrEmpty(config.OnCreateDirectoryExec))
+                return;
+
+            var directoryParameters = new DirectoryParameters
+            {
+                OnCreateDirectoryExec = config.OnCreateDirectoryExec,
+                OnCreateDirectoryExecArguments = config.OnCreateDirectoryExecArguments,
+                OnCreateDirectoryExecTimeout = config.OnCreateDirectoryExecTimeout.AsTimeSpan,
+                DatabaseName = databaseName,
+                Type = envType
+            };
+
+            void OnDirectory(StorageEnvironmentOptions internalOptions)
+            {
+                OnCreateDirectory(internalOptions, directoryParameters, logger);
+            }
+
+            options.OnCreateDirectory += OnDirectory;
+        }
+
+        public static void OnCreateDirectory(StorageEnvironmentOptions options, DirectoryParameters parameters, Logger log)
+        {
+            Process process = null;
+            try
+            {
+                var journalPath = string.Empty;
+                if (options is StorageEnvironmentOptions.DirectoryStorageEnvironmentOptions dirOptions)
+                    journalPath = dirOptions.JournalPath.FullPath;
+
+                var userArgs = parameters.OnCreateDirectoryExecArguments ?? string.Empty;
+                var args = $"{userArgs} {parameters.Type} {parameters.DatabaseName} " +
+                           $"{CommandLineArgumentEscaper.EscapeSingleArg(options.BasePath.ToString())} " +
+                           $"{CommandLineArgumentEscaper.EscapeSingleArg(options.TempPath.ToString())} " +
+                           $"{CommandLineArgumentEscaper.EscapeSingleArg(journalPath)}";
+
+                process = new Process
+                {
+                    StartInfo = new ProcessStartInfo
+                    {
+                        FileName = parameters.OnCreateDirectoryExec,
```

Comment:
We should check if this file exists

Reply:
Why? That would just give us a failure to start the process, which is already handled.